# This config file is used by ros2_control
controller_manager:
  ros__parameters:
    update_rate: 20  # Hz - Reduced to prevent overruns with 404ms CAN read time

    arm_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    gripper_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

arm_controller:
  ros__parameters:
    joints:
      - X_joint
      - Y_joint
      - Z_joint
      - A_joint
      - B_joint
      - C_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
    allow_nonzero_velocity_at_trajectory_end: true

gripper_controller:
  ros__parameters:
    joints:
      - Right_jaw_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
    allow_nonzero_velocity_at_trajectory_end: true